client/app/_services/user.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable } from '@angular/core';
+import { Response, URLSearchParams, Headers, RequestOptions }from '@angular/http';
+import { HttpClient } from './http.client';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/observable/of';
+
+import { User } from './userModel';
+@Injectable()
+export class UserService {
+public userData: any;
+public currentUser: any;
+public useradded: boolean;
+
+ constructor(private http:HttpClient ) { 
+  }
+  getUsers(): Observable<User[]>{
+    if(this.userData && this.useradded == false){
+      return Observable.of(this.userData);
```

Comment:
It should call the server to get the updated data always when the component of user list is displayed.